### ISSUE TITLE
Add `/credentials/issue` negative test for unknown issuer id

### ISSUE THREAD

**Issue by brownoxford** [w3c-ccg/traceability-interop#311]
If a system under test receives a `/credentials/issue` request that contains an issuer id for which the system does not have private key material, an error should be returned. Add a negative test for this scenario using `did:web:example.com` as the issuer ID, since it is not possible for any system to have private key material for that issuer ID.

Question: What should the expected error response be?

Suggestions:
- `400 Bad Request`
- `403 Forbidden`
- `500 Internal Server Error`

**Comment by nissimsan** [comment]
Go with `400 Bad Request`

**Comment by BenjaminMoe** [comment]
`400 Bad Request`

**Comment by brownoxford** [comment]
Blocked by #293

**Comment by TallTed** [comment]
I think `400` is not the right choice here, as that implies an error in the request, which is not the case; the issue is that the server lacks the private key material -- which presumably could be added to the server.

On that basis, I think [`501 Not Implemented`](https://www.rfc-editor.org/rfc/rfc9110.html#name-501-not-implemented) is the closest match, meaning `The server ... lacks the ability to fulfill the request` (though I think the textual portion might be adjusted).

_Maybe_ [`422 Unprocessable Content`](https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content) would be viable, if you can convince me that the client is in error, rather than the server being unable to handle the client's properly formed request --

>The 422 (Unprocessable Content) status code indicates that the server understands the content type of the request content (hence a [415 (Unsupported Media Type)](https://www.rfc-editor.org/rfc/rfc9110.html#status.415) status code is inappropriate), and the syntax of the request content is correct, but it was unable to process the contained instructions. For example, this status code can be sent if an XML request content contains well-formed (i.e., syntactically correct), but semantically erroneous XML instructions.

**Comment by brownoxford** [comment]
hmm. I see your point, but would lean towards 422 (if anything) because it is the client making the request with an incorrect value for issuer id. Servers publish valid values in their did:web documents, so it's reasonable to expect that the client uses a valid value here.

**Comment by TallTed** [comment]
@brownoxford -- Do you see no possibility of a server being incorrectly stood-up, such that it lacks some key material that it is *meant* to have, i.e., server admin error rather than client user error?

**Comment by brownoxford** [comment]
@TallTed I do, but I think that the best way to handle this one is to assume the server is set up correctly and select the best 4xx error. Otherwise we start getting into slippery slope territory with things like 400 errors, i.e., is it REALLY a 400, or did the admin accidentally delete the requested route and it should be a 5XX?

I agree that the initial choice of `400` is probably not best because the request is in fact syntactically correct, and that `422` may be a better option because "...the syntax of the request content is correct, but it was unable to process the contained instructions."

This is also a good opportunity for implementors to use the `details` portion of the error response to provide an indication of WHY the request cannot be satisfied.

I'm going to move this to a new issue so we can continue discussion.
